src/cursor/cur_layered.c

```diff
@@ -936,20 +904,31 @@ __layered_prev_int(WT_SESSION_IMPL *session, WT_CURSOR *cursor)
 static int
 __layered_prev(WT_CURSOR *cursor)
```

Comment:
I really like what you did with `__layered_prev` and `__layered_next` -- they're much easier to reason about now.